splitio/storage/pluggable.py

```diff
@@ -207,3 +207,170 @@ def put(self, split):
+        except Exception:
+            _LOGGER.error('Error updating segment change number')
+            _LOGGER.debug('Error: ', exc_info=True)
+
+    def get_change_number(self, segment_name):
+        """
+        Get a segment change number.
+
+        :param segment_name: segment name
+        :type segment_name: str
+
+        :return: change number
+        :rtype: int
+        """
+        try:
+            return self._pluggable_adapter.get(self._segment_till_prefix.format(segment_name=segment_name))
+        except Exception:
+            _LOGGER.error('Error fetching segment change number')
+            _LOGGER.debug('Error: ', exc_info=True)
+            return None
+
+    def get_segment_names(self):
+        """
+        Get list of segment names.
+
+        :return: list of segment names
+        :rtype: str[]
+        """
+        try:
+            keys = []
+            for key in self._pluggable_adapter.get_keys_by_prefix(self._prefix[:-self._SEGMENT_NAME_LEMNGTH]):
+                if key[-self._TILL_LENGTH:] != 'till':
+                    keys.append(key[len(self._prefix[:-self._SEGMENT_NAME_LEMNGTH]):])
+            return keys
+        except Exception:
+            _LOGGER.error('Error getting segments')
+            _LOGGER.debug('Error: ', exc_info=True)
+            return None
+
+    def get_keys(self, segment_name):
+        """
+        Get keys of a segment.
+
+        :param segment_name: segment name
+        :type segment_name: str
+
+        :return: list of segment keys
+        :rtype: str[]
+        """
+        try:
+            return list(self._pluggable_adapter.get(self._prefix.format(segment_name=segment_name)))
+        except Exception:
+            _LOGGER.error('Error getting segments keys')
+            _LOGGER.debug('Error: ', exc_info=True)
+            return None
+
+    def segment_contains(self, segment_name, key):
+        """
+        Check if segment contains a key
+
+        :param segment_name: segment name
+        :type segment_name: str
+        :param key: key
+        :type key: str
+
+        :return: True if found, otherwise False
+        :rtype: bool
+        """
+        try:
+            return self._pluggable_adapter.item_contains(self._prefix.format(segment_name=segment_name), key)
+        except Exception:
+            _LOGGER.error('Error checking segment key')
+            _LOGGER.debug('Error: ', exc_info=True)
+            return None
+
+    def get_segment_keys_count(self):
+        """
+        Get count of all keys in segments.
+
+        :return: keys count
+        :rtype: int
+        """
+        try:
+            return sum([self._pluggable_adapter.get_items_count(key) for key in self._pluggable_adapter.get_keys_by_prefix(self._prefix)])
+        except Exception:
+            _LOGGER.error('Error getting segment keys')
+            _LOGGER.debug('Error: ', exc_info=True)
+            return None
+
+    def get(self, segment_name):
```

Comment:
Do we need a `get` and `put` method? 
Up to you, consider keeping or removing them eventually if they are not used.

Reply:
The InMemory implementation has get and put, that is why I added them.